### PR TITLE
remove fake networking

### PR DESCRIPTION
**Pull request by barraguda**
## Problem

[kit PR](https://github.com/kinode-dao/kit/pull/140)

TODO:
- check chain for existing KNS name, and replace rather than register if true [X]
## Solution

{A brief description of how you solved the problem.}

## Docs Update

[Corresponding docs PR](https://github.com/kinode-dao/kinode-book/pull/my-pr-number)

## Notes

{Any other information useful for reviewers.}

### CODE REVIEW

File: kinode/src/main.rs [hunk 1]
```diff
@@ -3,19 +3,19 @@
 use anyhow::Result;
 use clap::{arg, value_parser, Command};
 use lib::types::core::*;
-#[cfg(feature = "simulation-mode")]
+//#[cfg(feature = "simulation-mode")]
```
Comment (dr-frmr) [inline kinode/src/main.rs]: Can remove this entirely instead of commenting out

File: kinode/src/main.rs [hunk 2]
```diff
@@ -3,19 +3,19 @@
...
 use ring::{rand::SystemRandom, signature, signature::KeyPair};
 use std::env;
 use std::sync::Arc;
 use tokio::sync::mpsc;
 
 mod eth;
+mod fakenet;
```
Comment (dr-frmr) [inline kinode/src/main.rs:14]: Should this be behind a feature flag? In general, please test compiling this PR in both simulation-mode and regularly, and make sure there are no compiler warnings about unused code in either compilation

File: kinode/src/net/mod.rs [hunk 3]
```diff
@@ -1,9 +1,6 @@
 #[cfg(feature = "simulation-mode")]
 pub mod mock;
```
Comment (dr-frmr) [inline kinode/src/net/mod.rs]: does `mock` still exist? why? I thought the code there was all obsolete with this

**Review by dr-frmr — COMMENTED** [review]
Left some comments, LMK

**Comment by nick1udwig** [comment]
I think [af877a0](https://github.com/kinode-dao/kinode/pull/318/commits/af877a03c7df8f3132b69a0d36356c8d928907f4) broke this PR. See https://github.com/kinode-dao/kit/pull/140#issuecomment-2087463067

File: kinode/src/main.rs (hunk 2 above)
Comment (barraguda) [reply kinode/src/main.rs:14]: I put it behind a flag for now even though I think there isn't a huge difference rn.

simulation-mode has a fair bit of previous and current unused variables in main.rs and register, we could do something like `#![cfg_attr(feature = "simulation-mode", allow(unused))]` in main.rs to surpress them..

File: kinode/packages/kns_indexer/kns_indexer/src/lib.rs [hunk 4]
```diff
@@ -112,16 +111,22 @@ fn subscribe_to_logs(eth_provider: &eth::Provider, from_block: u64, filter: eth:
...
+    let (chain_id, contract_address) = (31337, KNS_LOCAL_ADDRESS.to_string());
+    #[cfg(not(feature = "simulation-mode"))]
     let (chain_id, contract_address) = (10, KNS_OPTIMISM_ADDRESS.to_string());
-
     println!("indexing on contract address {}", contract_address);
 
+    #[cfg(feature = "simulation-mode")]
+    let kns_first_block: u64 = 1;
+    #[cfg(not(feature = "simulation-mode"))]
+    let kns_first_block: u64 = 114_923_786;
```
Comment (dr-frmr) [inline kinode/packages/kns_indexer/kns_indexer/src/lib.rs]: these can and should still be consts behind a feature flag

Comment (barraguda) [reply kinode/packages/kns_indexer/kns_indexer/src/lib.rs]: these are consts now, added ones for app_store too.

**Comment by barraguda** [comment]
a local app_store deployment is now also used in addition to the kns_indexer on simulation mode. I added a template package to the start state, but anvil currently does not store/replay past events if starting from a previously saved state. This should happen in the future though!

File: kinode/Cargo.toml [hunk 5]
```diff
@@ -88,3 +91,4 @@ wasi-common = "19.0.1"
 wasmtime = "19.0.1"
 wasmtime-wasi = "19.0.1"
 zip = "1.1.1"
+sha3 = "0.10.8"
```
Comment (dr-frmr) [inline kinode/Cargo.toml]: please place all cargo deps in alphabetical order

File: kinode/packages/kns_indexer/kns_indexer/src/lib.rs [hunk 6]
```diff
@@ -200,7 +208,8 @@ fn main(our: Address, mut state: State) -> anyhow::Result<()> {
                     }
                     break;
                 }
-                Err(_) => {
+                Err(e) => {
+                    println!("got some kind of error: {:?}", e);
                     println!("failed to fetch logs! trying again in 5s...");
```
Comment (dr-frmr) [inline kinode/packages/kns_indexer/kns_indexer/src/lib.rs]: can you merge these two error prints so they don't clog term too much?

File: kinode/src/main.rs (hunk 2 above)
Comment (dr-frmr) [reply kinode/src/main.rs:14]: It's best to keep unused code flagged-out for maximum clarity

**Review by dr-frmr — CHANGES_REQUESTED** [review]
Just a few final changes.

It's really exciting that any userspace program will be able to leverage the simulation-mode flag to enter a testing environment complete with their own contracts potentially.

Once you make final changes, please go in and assign flags in order to remove all unused code warnings (without using the allow-unused flag!). It helps to use mod { blocks } if there are a bunch of functions that are all sim-mode only or vice versa. I will check for this before merge

**Review by dr-frmr — APPROVED** [review]
LGTM!

**Review by dr-frmr — CHANGES_REQUESTED** [review]
Wait. Issue booting a real node.

**Review by dr-frmr — APPROVED** [review]
Working now for me.

**Pull request merged**